File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        int answer = 0;
        String str = "";
        String[] ms = myString.split("");
        
        for(int i=0; i<ms.length; i++){
            if(ms[i].equals("A")){
                str += "B";
            }else{
                str += "A";
            }
        }
        
        //if문 사용
        //if(str.contains(pat)){
        //    answer =1;
        //}
        //return answer;
       
        //삼항연산자 사용
        return str.contains(pat) ? 1 : 0;
    }
}


//다른사람 풀이
//class Solution {
//    public int solution(String myString, String pat) {
//        myString = myString.replace("A", "a").replace("B", "A").replace("a", "B");
//        return myString.contains(pat) ? 1 : 0;
//    }
//}

File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.js
function solution(myString, pat) {
    var answer = 0;
    var str = myString.split("");
    var str2 ="";
   
    for(var i=0; i<str.length; i++){
        if(str[i] === "A"){
            str[i] = "B"
        }else if(str[i] === "B"){
            str[i] = "A"
        }
        str2 += str[i];
        console.log('str2',str2);
    }
    
    if(str2.includes(pat)){
        answer=1;
    }
    return answer;
}